lib/HeaderSelector.php

```
@@ -54,72 +53,222 @@ public function selectHeaders($accept, $contentTypes)
      */
-    private function selectAcceptHeader($accept)
+    private function getHeaderAndWeight(string $header): array
     {
-        // Convert string to array if needed
-        if (is_string($accept)) {
-            $accept = [$accept];
+        # matches headers with weight, splitting the header and the weight in $outputArray
+        if (preg_match('/(.*);\s*q=(1(?:\.0+)?|0\.\d+)$/', $header, $outputArray) === 1) {
+            $headerData = [
+                'header' => $outputArray[1],
+                'weight' => (int)($outputArray[2] * 1000),
```

PHPStan (2.2.2)

[error] Binary operation "*" between non-falsy-string and 1000 results in an error.


(binaryOp.invalid)